.github/workflows/auto-create-release-pr.yml

```diff
@@ -0,0 +1,113 @@
+            exit 1
+          fi
+          echo "semver=${semver}" >> "$GITHUB_OUTPUT"
+
+          # Validate semver format X.Y.Z where X, Y, Z are numbers
+          if ! [[ "$semver" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            echo "Error: Invalid semver in branch name: $ref_name (extracted: $semver)" >&2
+            exit 1
+          fi
+
+          # Paginate to fetch all branches
+          all_branches="$(mktemp)"
+          page=1
+          total_pages=1  # Placeholder
+          echo "Fetching all branches (paginated)..."
+          while :; do
+            echo "Fetching page $page..."
+            resp="$(mktemp)"
+            url="https://api.github.com/repos/${GITHUB_REPOSITORY}/branches?per_page=100&page=${page}"
+            curl -sS -H "Authorization: token $GITHUB_TOKEN" -H "Accept: application/vnd.github.v3+json" "$url" -o "$resp"
+
+            # Append this page's branches to all_branches
+            cat "$resp" >> "$all_branches"
+
+            count=$(jq length "$resp")
+            if [ "$count" -lt 100 ]; then
+              total_pages=$page
+              break
+            fi
+            page=$((page + 1))
+          done
+          echo "Fetched $total_pages pages, total branches: $(jq -s 'map(length) | add' "$all_branches")"
+
+          # Process all branches: extract {name, semver} for matches, sort desc by semver
+          jq -s 'add | [ .[] | .name | select(test("^Version-v[0-9]+\\.[0-9]+\\.[0-9]+$") or test("^release/[0-9]+\\.[0-9]+\\.[0-9]+$")) | {name: ., semver: (if test("^Version-v") then ltrimstr("Version-v") else ltrimstr("release/") end) } ] | sort_by( .semver | split(".") | map(tonumber) ) | reverse' "$all_branches" > all_versions.json
+
+          # Print all found versions (sorted desc)
+          echo "All found versions (sorted desc): $(jq '[ .[].semver ]' all_versions.json)"
+
+          # Print all found matching branches (sorted by semver desc)
+          echo "All found matching branches:"
+          jq -r '.[].name' all_versions.json || echo "No matching branches found."
+
+          # Filter to those with semver strictly lower than current
+          jq --arg semver "$semver" '[ .[] | select( .semver as $v | $semver | split(".") as $c | $v | split(".") as $p | ($p[0] | tonumber) < ($c[0] | tonumber) or (($p[0] | tonumber) == ($c[0] | tonumber) and (($p[1] | tonumber) < ($c[1] | tonumber) or (($p[1] | tonumber) == ($c[1] | tonumber) and ($p[2] | tonumber) < ($c[2] | tonumber)))) ) ]' all_versions.json > filtered_versions.json
+
+          # Print filtered versions (still sorted desc, so [0] is highest lower)
+          echo "Filtered versions (< $semver, sorted desc): $(jq '[ .[].semver ]' filtered_versions.json)"
+
+          # Select the highest lower: first in filtered list
+          if [ "$(jq length filtered_versions.json)" -eq 0 ]; then
+            previous_ref="main"
```

Comment:
I don't think we shall fall back to "main" when no highest lower was found. I'd prefer to the entire workflow to fail when that happens as it's something that's not supposed to happen.